Failed to shutdown the HTTP server for OAuth flow, when using sockets + customRoutes

### Description

Using `socketMode = true` with `customRoutes` (simple health check from documentation in fact) results in `[ERROR] Failed to shutdown the HTTP server for OAuth flow: undefined` on `app.stop()`

The reason (in my opinion) is this code in `SocketModeReceiver.js`:
```js
    stop() {
        if (this.httpServer !== undefined) {
            // This HTTP server is only for the OAuth flow support
            this.httpServer.close((error) => {
                this.logger.error(`Failed to shutdown the HTTP server for OAuth flow: ${error}`);
                ^^^^^
            });
        }
```
Here is no condition to check if error occured or not
The fix would be:
```js
if (error) this.logger.error(`Failed to shutdown the HTTP server for OAuth flow: ${error}`);
```

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [X] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [X] I've read and understood the [Contributing guidelines](https://github.com/slackapi/bolt/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [X] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

package version:

node version:

OS version(s):

#### Steps to reproduce:

1.
2.
3.

#### Expected result:

What you expected to happen

#### Actual result:

What actually happened

#### Attachments:

Logs, screenshots, screencast, sample project, funny gif, etc.

--- a/src/receivers/SocketModeReceiver.ts
+++ b/src/receivers/SocketModeReceiver.ts
@@ -241,7 +241,7 @@ export default class SocketModeReceiver implements Receiver {
     if (this.httpServer !== undefined) {
       // This HTTP server is only for the OAuth flow support
       this.httpServer.close((error) => {
-        this.logger.error(`Failed to shutdown the HTTP server for OAuth flow: ${error}`);
+        if (error) this.logger.error(`Failed to shutdown the HTTP server for OAuth flow: ${error}`);
       });
     }
     return new Promise((resolve, reject) => {

